### PR TITLE
SW-4516 Refactor InventorySeedlingsTable for use in two views

### PR DESCRIPTION
**Pull request by nickgraz**
- Used in both 'inventory by species' and 'inventory by nursery' views
- Rename a few files for consistency
- Allows us to build more functionality into the tables with less duplication
- Separates the origin-specific logic from the UI

### CODE REVIEW

File: src/components/InventoryV2/view/InventorySeedlingsTableForNursery.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  { key: 'totalQuantity', name: strings.TOTAL, type: 'string' },
+  { key: 'totalQuantityWithdrawn', name: strings.WITHDRAWN, type: 'string' },
+  { key: 'readyByDate', name: strings.EST_READY_DATE, type: 'date' },
+  { key: 'addedDate', name: strings.DATE_ADDED, type: 'date' },
+  { key: 'withdraw', name: '', type: 'string' },
+  { key: 'quantitiesMenu', name: '', type: 'string' },
+];
+
+export default function InventorySeedlingsTableForNursery(props: InventorySeedlingsTableForNurseryProps): JSX.Element {
+  const facilityId = props.nurseryId;
```
Comment (nickgraz) [inline src/components/InventoryV2/view/InventorySeedlingsTableForNursery.tsx:31]: This is done with the intention that we might use the `InventorySeedlingsTable` for other facility types in the future.

File: src/components/InventoryV2/view/InventorySeedlingsTable.tsx [hunk 2]
```diff
@@ -414,7 +428,7 @@ export default function InventorySeedlingsTable(props: InventorySeedlingsTablePr
           <Box marginTop={theme.spacing(2)}>
             <Table
               id='inventory-seedlings-table'
```
Comment (karthikbtf) [inline src/components/InventoryV2/view/InventorySeedlingsTable.tsx:430]: Make this `{`inventory-seedlings-table-${origin.toLowerCase()}`}` so we have unique table ids by use-case.
The id is used as the preference name to store table column ordering, if user shuffles the columns.

**Review by karthikbtf — COMMENTED** [review]
Nice refactor, one comment so far..

File: src/components/InventoryV2/view/InventorySeedlingsTableForNursery.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    (
+      orgId: number,
+      originId: number,
+      searchFields: FieldNodePayload[],
+      searchSortOrder: SearchSortOrder | undefined
+    ): Promise<SearchResponseElement[] | null> => {
+      // TODO it seems this was not implemented
+      // tslint:disable-next-line:no-console
+      console.log(
+        'TODO - implement batches export for inventory by nursery',
```
Comment (constanzauanini) [inline src/components/InventoryV2/view/InventorySeedlingsTableForNursery.tsx:73]: I don't see this export in designs. For species either. I think it's replaced with the exports of the lists tables

Comment (nickgraz) [reply src/components/InventoryV2/view/InventorySeedlingsTableForNursery.tsx:73]: I see, OK I will just implement it as a noop for the nursery view

**Comment by karthikbtf** [comment]
Looks good, nice refactor.
The column names don't seem to be rendering.

Terraware App 2023-11-29 09-54-16

**Comment by nickgraz** [comment]
> Looks good, nice refactor. The column names don't seem to be rendering.
> 
> Terraware App 2023-11-29 09-54-16

Hmm... the `strings` stopped working. Should be an easy fix 
SCR-20231129-kbmx

**Review by karthikbtf — APPROVED** [review]
Nice!

**Pull request merged**